src/main/java/org/intellij/sdk/codesync/CodeSyncSetup.java

```diff
@@ -0,0 +1,499 @@
+
+            if (!configFile.isRepoSynced(repoPath) || !repo.isSuccessfullySynced()) {
+
+                // Do not ask user to sync repo, if it is already in progress.
+                if (!reposBeingSynced.contains(repoPath)) {
+                    reposBeingSynced.add(repoPath);
+                    boolean shouldSyncRepo = Messages.showYesNoDialog(
+                            "Do you want to enable syncing of this Repo?",
+                            String.format("'%s' Is not Being Synced!", repoName),
+                            Notification.YES,
+                            Notification.NO,
+                            Messages.getQuestionIcon()
+                    ) == Messages.YES;;
+
+                    if (shouldSyncRepo) {
+                        boolean hasAccessToken = checkUserAccess();
+                        if (hasAccessToken) {
+                            syncRepoAsync(repoPath, repoName, Utils.GetGitBranch(repoPath));
+                        }
+                    }
+                }
+            }
+
+        } catch (InvalidConfigFileError error) {
+            CodeSyncLogger.logEvent(String.format("Config file error, %s.\n", error.getMessage()));
+        }
+
+        // Create system directories required by the plugin.
+        createSystemDirectories();
+    }
+
+    public static void createSystemDirectories() {
+        // Create system folders
+        String[] systemFolders = {CODESYNC_ROOT, DIFFS_REPO, ORIGINALS_REPO, SHADOW_REPO, DELETED_REPO};
+        for (String systemFolder : systemFolders) {
+            File folder = new File(systemFolder);
+            folder.mkdirs();
+        }
+    }
+
+    /*
+    Check if the user has proper access for repo initialization, or not.
+
+    This will also trigger the authentication flow if user does not have access token setup.
+     */
+    public static boolean checkUserAccess() {
+        CodeSyncClient codeSyncClient = new CodeSyncClient();
+        if (!codeSyncClient.isServerUp()) {
+            NotificationManager.notifyError(Notification.SERVICE_NOT_AVAILABLE);
+            return false;
+        }
+
+        String accessToken = null;
+        try {
+            UserFile userFile = new UserFile(USER_FILE_PATH);
+            UserFile.User user = userFile.getUser();
+            if (user != null) {
+                accessToken = user.getAccessToken();
+            }
+        } catch (FileNotFoundException | InvalidYmlFileError error) {
+            // Set access token to null to trigger user signup.
+            accessToken = null;
+        }
+
+        // User already has access token, no need to proceed.
+        if (accessToken != null) {
+            try {
+                Pair<Boolean, User> userPair = codeSyncClient.getUser(accessToken);
+                Boolean isTokenValid = userPair.component1();
+
+                // If token is not valid then need to authenticate again.
+                if (!isTokenValid) {
+                    return false;
```

Comment:
I think need to trigger Login Flow at this point (maybe add a TODO for now)

Reply:
Done